master/static/migrations/20220528215116_reindex-raw-steps.tx.up.sql

```diff
@@ -0,0 +1,82 @@
+ALTER TABLE raw_steps DROP CONSTRAINT steps_trial_id_run_id_total_batches_unique;
+-- this index is  UNIQUE CONSTRAINT, btree (trial_id, trial_run_id, total_batches)
+-- almost all trial_id=1, so having trial_run_id before total batches 
+-- discourages the planner from using index for total_batches
+--
+--
+-- e.g. LEFT JOIN raw_steps s ON ((c.metadata ->> 'steps_completed'::text)::integer) = s.total_batches AND t.id = s.trial_id
+-- results in
+--
+  -- "Hash Cond": "((s.total_batches = ((c_1.metadata ->> 'steps_completed'::text))::integer) AND (s.trial_id = t.id))",
+  -- "Plans": [
+  --   {
+  --     "Node Type": "Seq Scan",
+  --     "Parent Relationship": "Outer",
+  --     "Parallel Aware": false,
+  --     "Relation Name": "raw_steps",
+  --     "Schema": "public",
+  --     "Alias": "s",
+  --     "Startup Cost": 0,
+  --     "Total Cost": 1197184.22,
+  --     "Plan Rows": 719522,
+  --     "Plan Width": 825,
+  --     "Actual Startup Time": 0.033,
+  --     "Actual Total Time": 51451.589,
+  --     "Actual Rows": 720038,
+  --     "Actual Loops": 1,
+  --     "Output": ["s.metrics", "s.total_batches", "s.trial_id", "s.archived"]
+  --   }
+  -- ]
+
+ALTER TABLE raw_steps CREATE UNIQUE INDEX steps_trial_id_total_batches_end_time_run_id ON raw_steps(trial_id, total_batches, trial_run_id);
+-- the unique constraint here is NOT equivalent to the previous unique contraint
+-- since it would allow records with same (trial_id, trial_run_id, total_batches)
+-- but different end_times. 
+
+-- if we do need the constraint, we would probably be better off putting it separately
+
```

Comment:
instead of replace index fields, could also try reordering them to  `(trial_id, total_batches, trial_run_id)`

and see how it affects performance. based on where `trial_run_id` appears, i dont think it could hurt anything